src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp

```diff
@@ -178,8 +178,21 @@ namespace OpenMS
     connect(&tab_bar_, &EnhancedTabBar::dropOnTab, this, &TOPPViewBase::copyLayer);
     box_layout->addWidget(&tab_bar_);
 
+    //Trigger updates only when the active subWindow index changes and update it
     connect(&ws_, &EnhancedWorkspace::subWindowActivated, [this](QMdiSubWindow* window) {
-      if (window != nullptr) /* 0 upon terminate */ updateBarsAndMenus(); 
+      Size s = 0;
+      int oldactiveSubwindowID = activeSubwindowID_;
```

Comment:
why not store the pointer to the window directly?
This would safe us from iterating through the window list and one can simply do:
```
if (window == nullptr) return; /* 0 upon terminate */ 
if (active_subwindow_ != window)
{
  active_subwindow_ == window;
  updateBarsAndMenus();
}
```

Reply:
I think we need it later to get the currently _visible_ plot.
See my other comment. I'm not 100% sure anymore. I was and still am a bit confused what Qt means with "active" and if the MDI case if different. Sounds like it in the Qt docu.